docs: refresh testing_integration.rst

goneri opened the pull request:
Ensures the examples can be run:

- fedora25 is not available anymore, use fedora29 instead
- the posix/ci alias does not exist anymore, use shippable/posix/
  instead
- explain how to list the target without argcomplete

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

testing_integration.rst

goneri committed:
docs: refresh testing_integration.rst

Ensures the examples can be run:

- fedora25 is not available anymore, use fedora29 instead
- the posix/ci alias does not exist anymore, use shippable/posix/
  instead
- explain how to list the target without argcomplete

mattclay commented on docs/docsite/rst/dev_guide/testing_integration.rst:
```diff
@@ -113,7 +117,7 @@ Running Integration Tests
 To run all CI integration test targets for POSIX platforms in a Ubuntu 16.04 container::
 
-    test/runner/ansible-test integration -v posix/ci/ --docker
+    test/runner/ansible-test integration --docker -v shippable/
```
```suggestion
    test/runner/ansible-test integration --docker ubuntu1604 -v shippable/
```

mattclay commented on docs/docsite/rst/dev_guide/testing_integration.rst:
```diff
@@ -55,23 +55,27 @@ outside of those test subdirectories.  They will also not reconfigure or bounce
 Run as follows for all POSIX platform tests executed by our CI system::
 
-    test/runner/ansible-test integration --docker fedora25 -v posix/ci/
+    test/runner/ansible-test integration --docker fedora29 -v shippable/posix/
```
The `ansible-test` invocations could be simplified if the docs mentioned sourcing `hacking/env-setup`, since that puts `ansible-test` in `PATH`.

```suggestion
    ansible-test integration --docker fedora29 -v shippable/
```

mattclay commented on docs/docsite/rst/dev_guide/testing_integration.rst:
```diff
@@ -33,7 +33,7 @@ more information about supported credentials, refer to ``credentials.template``.
 =============
 
-The tests will assume things like hg, svn, and git are installed and in path.  Some tests
+The tests will assume things like hg, svn, and git are installed, and in path.  Some tests
```
```suggestion
Some tests assume things like hg, svn, and git are installed, and in path.  Some tests
```

mattclay committed:
Apply suggestions from code review

Co-Authored-By: goneri <goneri@lebouder.net>

mattclay commented on docs/docsite/rst/dev_guide/testing_integration.rst:
```diff
@@ -55,23 +55,27 @@ outside of those test subdirectories.  They will also not reconfigure or bounce
 Run as follows for all POSIX platform tests executed by our CI system::
 
-    test/runner/ansible-test integration --docker fedora25 -v posix/ci/
+    ansible-test integration --docker fedora29 -v shippable/
```
Sorry, I should have been more clear on my previous feedback. You'll probably want to add a brief section mentioning use of `source hacking/env-setup` (as is found in several other places in our docs, although not currently in this file), so that `ansible-test` is in the path. Then update this example, as you've done, along with all the others in this file to use `ansible-test` instead of `test/runner/ansible-test`.

goneri replied:
Is it better this way: https://github.com/ansible/ansible/pull/53338/commits/205b960a662e6fccb9747605cfdf3a3d483c97b3

As way tempted to just add `source hacking/env-setup` before every ansible-test example. This may be enough.

goneri committed:
ansible-test should be in $PATH

In order to simplify the example, we assume `ansible-runner` is in the $PATH.
This patch explains how to get this done.

mattclay commented on docs/docsite/rst/dev_guide/testing_integration.rst:
```diff
@@ -25,6 +25,22 @@ It provides tab completion in ``bash`` for the ``ansible-test`` test runner.
+ansible-test command
+--------------------
+
+The example below assumes ``test/runner`` is in your ``$PATH``. An easy way to achieve that
```
```suggestion
The example below assumes ``bin`` is in your ``$PATH``. An easy way to achieve that
```

This was written before `ansible-test` was added to `bin`. At the time it was only available as `test/runner/ansible-test`, but is now also available from the preferred location of `bin/ansible-test`. Sourcing `hacking/env-setup` makes the one under `bin` available.

mattclay commented on docs/docsite/rst/dev_guide/testing_integration.rst:
```diff
@@ -25,6 +25,22 @@ It provides tab completion in ``bash`` for the ``ansible-test`` test runner.
+
+You can also call ``ansible-test`` with the full path::
+
+    test/runner/ansible-test --help
```
```suggestion
    bin/ansible-test --help
```

goneri committed:
call ansible-runner from bin/, not test/runner

Since f3d1f9544ba67785e787ea303f81db74603780eb, `bin/` is now the
new home of `ansible-runner`.

mattclay commented on docs/docsite/rst/dev_guide/testing_integration.rst:
```diff
@@ -55,23 +71,27 @@ outside of those test subdirectories.  They will also not reconfigure or bounce
+.. note:: Bash users
 
-    test/runner/ansible-test integration <tab><tab>
+   If you use ``Bash`` with ``argcomplete``, obtain a full list by doing: ``test/runner/ansible-test integration <tab><tab>``
```
```suggestion
   If you use ``bash`` with ``argcomplete``, obtain a full list by doing: ``ansible-test integration <tab><tab>``
```

mattclay commented on docs/docsite/rst/dev_guide/testing_integration.rst:
```diff
@@ -94,7 +114,7 @@ Define Windows inventory::
 Run the Windows tests executed by our CI system::
 
-    test/runner/ansible-test windows-integration -v windows/ci/
+    ansible-test windows-integration -v windows/ci/
```
```suggestion
    ansible-test windows-integration -v shippable/
```

mattclay commented on docs/docsite/rst/dev_guide/testing_integration.rst:
```diff
@@ -25,6 +25,22 @@ It provides tab completion in ``bash`` for the ``ansible-test`` test runner.
+
+integration_config.yml
+----------------------
+
```
The section below that references `credentials.template` is out-of-date as well. I'd probably replace the last sentence:

> For more information about supported credentials, refer to ``credentials.template``.

with:

> For more information about supported credentials, refer to the various ``cloud-config-*.template`` files in the ``test/integration/`` directory.

goneri committed:
we don't use credentials.template anymore

The file is now replaced by a collection of .template files.

goneri committed:
low-case bash bc we refresh to name of the binary

goneri committed:
refresh the command of Windows-CI example

mattclay commented on docs/docsite/rst/dev_guide/testing_integration.rst:
```diff
@@ -55,23 +72,27 @@ outside of those test subdirectories.  They will also not reconfigure or bounce
+.. note:: Bash users
 
-    test/runner/ansible-test integration <tab><tab>
+   If you use ``bash`` with ``argcomplete``, obtain a full list by doing: ``test/runner/ansible-test integration <tab><tab>``
```
```suggestion
   If you use ``bash`` with ``argcomplete``, obtain a full list by doing: ``ansible-test integration <tab><tab>``
```

mattclay committed:
Update docs/docsite/rst/dev_guide/testing_integration.rst

Co-Authored-By: goneri <goneri@lebouder.net>

acozine commented:
Thanks @goneri for updating the instructions on running the tests. It's great to get a fresh set of eyes (and a full test) on that page!
